### PR TITLE
Add OpenStack Identity Role Mapping module

### PR DESCRIPTION
**Pull request by emonty**

### CODE REVIEW

File: cloud/openstack/os_user_role.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+'''
+
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+    argument_spec = dict(
+        role=dict(required=True),
+        user=dict(required=True),
+        group=dict(required=True),
```
Comment (omgjlk) [inline cloud/openstack/os_user_role.py:81]: Docs say that user /or/ group are required, but both are listed as required here. Which is right?

Reading below, both should be marked as not required, as you take care of that on line 89.

File: cloud/openstack/os_user_role.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            if assignment:
+                cloud.revoke_role(
+                    role_name_or_id=role,
+                    user_name_or_id=user,
+                    domain_name_or_id=domain,
+                    project_name_or_id=project)
+                changed=True
+            else:
+                changed=False
+        module.exit_json(changed=changed)
```
Comment (omgjlk) [inline cloud/openstack/os_user_role.py:138]: Seems like check mode could be implemented pretty easily here.

**Comment by gregdek** [comment]
@emonty ping for update

**Comment by gregdek** [comment]
@emonty @shrews @dguerri @juliakreger @j2sol @rcarrillocruz ping, any updates here?

**Comment by gregdek** [comment]
@emonty A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by robynbergeron** [comment]
@emonty Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If we don't hear from you within another 14 days, we will close this pull request.

**Comment by gregdek** [comment]
Since this module's been waiting on revisions for six months, I'm gonna go ahead and close this PR. @emonty and friends, feel free to reopen or submit a new PR if you see fit.

**Pull request closed without merging**